brave/src/main/java/brave/propagation/ExtraFieldPropagation.java

```diff
@@ -378,19 +454,23 @@ public static void set(TraceContext context, String name, String value) {
       this.setter = setter;
     }
 
-    @Override public void inject(TraceContext traceContext, C carrier) {
-      delegate.inject(traceContext, carrier);
-      Extra extra = traceContext.findExtra(Extra.class);
+    @Override public void inject(TraceContext ctx, C carrier) {
+      delegate.inject(ctx, carrier);
+      Extra extra = ctx.findExtra(Extra.class);
       if (extra == null) return;
-      inject(extra, carrier);
+      inject(extra, carrier, injectCustomizers(propagation.factory.customizers, ctx));
     }
 
-    void inject(Extra fields, C carrier) {
-      for (int i = 0, length = propagation.keys.size(); i < length; i++) {
-        if (propagation.redacted.get(i)) continue; // don't propagate downstream
-        String maybeValue = fields.get(propagation.factory.keyToField[i]);
+    void inject(Extra fields, C carrier, FieldCustomizer[] fieldCustomizers) {
+      for (int i = 0, length = propagation.keys.length; i < length; i++) {
+        int j = propagation.factory.keyToField[i];
+        String fieldName = propagation.factory.fieldNames[j];
+        String maybeValue = fields.get(j);
+        for (FieldCustomizer fieldCustomizer : fieldCustomizers) {
```

Comment:
interally this uses arrays to cheapen the extra work we will now do on every inject/extract. This is especially cheap as it will never allocate an iterator, yet we can still use for each.